From 49b04e4036ce40d0a1481fc28cd0d86b3c70493a Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Tue, 31 Jul 2012 18:44:20 +0300
Subject: [PATCH 279/370] conf-map-ut: new test suite

+ utils/ut_main.c: sort unit tests alphabetically, in accordance with
  the comments.
+ Replace tabs with spaces in several places, for uniformity.
---
 Makefile.am         |  9 +++----
 conf/ut/Makefile.am |  2 ++
 conf/ut/map.c       |  7 +++---
 configure.ac        | 35 ++++++++++++++--------------
 utils/Makefile.am   |  3 ++-
 utils/ut_main.c     | 67 +++++++++++++++++++++++++++--------------------------
 6 files changed, 65 insertions(+), 58 deletions(-)
 create mode 100644 conf/ut/Makefile.am

diff --git a/Makefile.am b/Makefile.am
index 7975f56..3b5f99d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -11,7 +11,6 @@ SUBDIRS_LIBS =                     \
                fol                 \
                db                  \
                fid                 \
-                                   \
                addb/addbff         \
                net                 \
                fop                 \
@@ -38,14 +37,16 @@ SUBDIRS_LIBS =                     \
                ut                  \
                xcode               \
                yaml2db
+
 SUBDIR_IT = rpc/it
 
-SUBDIR_UT = fop/ut fop/ut/long_lock lib/ut stob/ut layout/ut balloc/ut    \
-	    db/ut fol/ut sns/ut desim/ut cob/ut console/ut colibri/ut     \
-	    net/ut net/bulk_emulation/ut net/lnet/ut udb/ut capa/ut       \
+SUBDIR_UT = fop/ut fop/ut/long_lock lib/ut stob/ut layout/ut balloc/ut \
+	    db/ut fol/ut sns/ut desim/ut cob/ut conf/ut console/ut colibri/ut \
+	    net/ut net/bulk_emulation/ut net/lnet/ut udb/ut capa/ut \
 	    ioservice/ut reqh/ut rpc/ut sm/ut xcode/ut yaml2db/ut
 
 SUBDIR_KERNEL = build_kernel_modules
+
 SUBDIR_ST = console/st net/bulk_emulation/st net/lnet/st yaml2db/st
 
 SUBDIR_UB =
diff --git a/conf/ut/Makefile.am b/conf/ut/Makefile.am
new file mode 100644
index 0000000..d63bcbf
--- /dev/null
+++ b/conf/ut/Makefile.am
@@ -0,0 +1,2 @@
+noinst_LTLIBRARIES    = libconf-ut.la
+libconf_ut_la_SOURCES = map.c
diff --git a/conf/ut/map.c b/conf/ut/map.c
index 4cd7269..b6bf3b2 100644
--- a/conf/ut/map.c
+++ b/conf/ut/map.c
@@ -55,12 +55,13 @@ static void test_add_lookup(void)
 	c2_conf_map_fini(&map);
 }
 
-const struct c2_test_suite c2_conf_map_ut = {
-	.ts_name  = "conf_map",
+const struct c2_test_suite conf_map_ut = {
+	.ts_name  = "conf-map-ut",
 	.ts_init  = NULL,
 	.ts_fini  = NULL,
 	.ts_tests = {
-		{ "add & lookup", test_add_lookup },
+		{ "conf-map-add-lookup", test_add_lookup },
 		{ NULL, NULL }
 	}
 };
+C2_EXPORTED(conf_map_ut);
diff --git a/configure.ac b/configure.ac
index ce8ce91..0e00c6b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -639,8 +639,8 @@ AC_SUBST([KCFLAGS])
 AC_CONFIG_FILES([
                  Makefile
                  addb/Makefile
-		 addb/addbff/Makefile
-		 addb/ut/Makefile
+                 addb/addbff/Makefile
+                 addb/ut/Makefile
                  balloc/Makefile
                  balloc/ut/Makefile
                  build_kernel_modules/Makefile
@@ -653,7 +653,8 @@ AC_CONFIG_FILES([
                  cob/ut/Makefile
                  colibri/Makefile
                  colibri/ut/Makefile
-		 console/Makefile
+                 conf/ut/Makefile
+                 console/Makefile
                  console/bin/Makefile
                  console/ut/Makefile
                  console/st/Makefile
@@ -681,23 +682,23 @@ AC_CONFIG_FILES([
                  mds/Makefile
                  mw/Makefile
                  net/Makefile
-		 net/bulk_emulation/Makefile
-		 net/bulk_emulation/st/Makefile
-		 net/bulk_emulation/ut/Makefile
-		 net/lnet/Makefile
-		 net/lnet/st/Makefile
-		 net/lnet/st/linux_kernel/Makefile
-		 net/lnet/ut/Makefile
-		 net/ut/Makefile
+                 net/bulk_emulation/Makefile
+                 net/bulk_emulation/st/Makefile
+                 net/bulk_emulation/ut/Makefile
+                 net/lnet/Makefile
+                 net/lnet/st/Makefile
+                 net/lnet/st/linux_kernel/Makefile
+                 net/lnet/ut/Makefile
+                 net/ut/Makefile
                  nrs/Makefile
                  patches/Makefile
                  pool/Makefile
                  reqh/Makefile
-		 reqh/ut/Makefile
+                 reqh/ut/Makefile
                  rpc/Makefile
                  rpc/ut/Makefile
                  rpc/it/Makefile
-		 rpc/it/linux_kernel/Makefile
+                 rpc/it/linux_kernel/Makefile
                  sm/Makefile
                  sm/ut/Makefile
                  sns/Makefile
@@ -716,10 +717,10 @@ AC_CONFIG_FILES([
                  xcode/Makefile
                  xcode/ff2c/Makefile
                  xcode/ut/Makefile
-		 yaml2db/Makefile
-		 yaml2db/ut/Makefile
-		 yaml2db/st/Makefile
-		 colibri.spec
+                 yaml2db/Makefile
+                 yaml2db/ut/Makefile
+                 yaml2db/st/Makefile
+                 colibri.spec
                 ])
 AC_OUTPUT
 
diff --git a/utils/Makefile.am b/utils/Makefile.am
index c738fcb..415a76d 100644
--- a/utils/Makefile.am
+++ b/utils/Makefile.am
@@ -28,7 +28,8 @@ ut_LDADD     = $(top_builddir)/colibri/libcolibri.la \
                $(top_builddir)/yaml2db/ut/libyaml2db-ut.la \
 	       $(top_builddir)/addb/ut/libaddb-ut.la \
 	       $(top_builddir)/balloc/ut/libballoc-ut.la \
-	       $(top_builddir)/fop/ut/long_lock/libfop-lock-ut.la
+	       $(top_builddir)/fop/ut/long_lock/libfop-lock-ut.la \
+	       $(top_builddir)/conf/ut/libconf-ut.la
 
 ub_SOURCES   = ub_main.c common.c common.h
 ub_LDFLAGS   = -pthread
diff --git a/utils/ut_main.c b/utils/ut_main.c
index cf4ffc0..376d357 100644
--- a/utils/ut_main.c
+++ b/utils/ut_main.c
@@ -30,26 +30,36 @@
 
 /* sort test suites in alphabetic order */
 extern const struct c2_test_suite libc2_ut; /* test lib first */
-extern const struct c2_test_suite adieu_ut;
 extern const struct c2_test_suite ad_ut;
 extern const struct c2_test_suite addb_ut;
-extern const struct c2_test_suite bulkio_server_ut;
+extern const struct c2_test_suite adieu_ut;
+extern const struct c2_test_suite balloc_ut;
+extern const struct c2_test_suite buffer_pool_ut;
 extern const struct c2_test_suite bulkio_client_ut;
-extern const struct c2_test_suite ios_bufferpool_ut;
-extern const struct c2_test_suite cobfoms_ut;
+extern const struct c2_test_suite bulkio_server_ut;
+extern const struct c2_test_suite c2_fop_lock_ut;
+extern const struct c2_test_suite c2_net_bulk_if_ut;
+extern const struct c2_test_suite c2_net_bulk_mem_ut;
+extern const struct c2_test_suite c2_net_lnet_ut;
+extern const struct c2_test_suite c2_net_tm_prov_ut;
 extern const struct c2_test_suite capa_ut;
+extern const struct c2_test_suite cfm_ut;
 extern const struct c2_test_suite cob_ut;
+extern const struct c2_test_suite cobfoms_ut;
+extern const struct c2_test_suite colibri_setup_ut;
+extern const struct c2_test_suite conf_reg_ut;
 extern const struct c2_test_suite console_ut;
-extern const struct c2_test_suite db_ut;
 extern const struct c2_test_suite db_cursor_ut;
+extern const struct c2_test_suite db_ut;
 extern const struct c2_test_suite emap_ut;
 extern const struct c2_test_suite fit_ut;
 extern const struct c2_test_suite fol_ut;
-extern const struct c2_test_suite c2_net_bulk_if_ut;
-extern const struct c2_test_suite c2_net_bulk_mem_ut;
-extern const struct c2_test_suite c2_net_lnet_ut;
-extern const struct c2_test_suite c2_net_tm_prov_ut;
+extern const struct c2_test_suite frm_ut;
+extern const struct c2_test_suite ios_bufferpool_ut;
 extern const struct c2_test_suite parity_math_ut;
+extern const struct c2_test_suite reqh_ut;
+extern const struct c2_test_suite rpc_service_ut;
+extern const struct c2_test_suite rpclib_ut;
 extern const struct c2_test_suite sm_ut;
 extern const struct c2_test_suite stobio_ut;
 extern const struct c2_test_suite udb_ut;
@@ -57,58 +67,49 @@ extern const struct c2_test_suite xcode_bufvec_fop_ut;
 extern const struct c2_test_suite xcode_bufvec_ut;
 extern const struct c2_test_suite xcode_ff2c_ut;
 extern const struct c2_test_suite xcode_ut;
-extern const struct c2_test_suite reqh_ut;
-extern const struct c2_test_suite colibri_setup_ut;
-extern const struct c2_test_suite rpclib_ut;
-extern const struct c2_test_suite cfm_ut;
 extern const struct c2_test_suite yaml2db_ut;
-extern const struct c2_test_suite buffer_pool_ut;
-extern const struct c2_test_suite addb_ut;
-extern const struct c2_test_suite balloc_ut;
-extern const struct c2_test_suite rpc_service_ut;
-extern const struct c2_test_suite frm_ut;
-extern const struct c2_test_suite c2_fop_lock_ut;
 
 #define UT_SANDBOX "./ut-sandbox"
 
 void add_uts(void)
 {
 	/* sort test suites in alphabetic order */
-	c2_ut_add(&libc2_ut);
+	c2_ut_add(&libc2_ut); /* test lib first */
 	c2_ut_add(&ad_ut);
 	c2_ut_add(&adieu_ut);
 	c2_ut_add(&balloc_ut);
 	c2_ut_add(&buffer_pool_ut);
-        c2_ut_add(&bulkio_server_ut);
-        c2_ut_add(&bulkio_client_ut);
-        c2_ut_add(&ios_bufferpool_ut);
+	c2_ut_add(&bulkio_client_ut);
+	c2_ut_add(&bulkio_server_ut);
+	c2_ut_add(&c2_fop_lock_ut);
+	c2_ut_add(&c2_net_bulk_if_ut);
+	c2_ut_add(&c2_net_bulk_mem_ut);
+	c2_ut_add(&c2_net_lnet_ut);
+	c2_ut_add(&c2_net_tm_prov_ut);
 	c2_ut_add(&capa_ut);
 	c2_ut_add(&cfm_ut);
 	c2_ut_add(&cob_ut);
-        c2_ut_add(&cobfoms_ut);
+	c2_ut_add(&cobfoms_ut);
 	c2_ut_add(&colibri_setup_ut);
-	c2_ut_add(&db_ut);
+	c2_ut_add(&conf_reg_ut);
 	c2_ut_add(&db_cursor_ut);
+	c2_ut_add(&db_ut);
 	c2_ut_add(&emap_ut);
 	c2_ut_add(&fit_ut);
 	c2_ut_add(&fol_ut);
-	c2_ut_add(&c2_fop_lock_ut);
-	c2_ut_add(&c2_net_bulk_if_ut);
-	c2_ut_add(&c2_net_bulk_mem_ut);
-	c2_ut_add(&c2_net_lnet_ut);
-	c2_ut_add(&c2_net_tm_prov_ut);
-	c2_ut_add(&parity_math_ut);
 	c2_ut_add(&frm_ut);
+	c2_ut_add(&ios_bufferpool_ut);
+	c2_ut_add(&parity_math_ut);
 	c2_ut_add(&reqh_ut);
-	c2_ut_add(&rpclib_ut);
 	c2_ut_add(&rpc_service_ut);
+	c2_ut_add(&rpclib_ut);
 	c2_ut_add(&sm_ut);
 	c2_ut_add(&stobio_ut);
 	c2_ut_add(&udb_ut);
 	c2_ut_add(&xcode_bufvec_fop_ut);
 	c2_ut_add(&xcode_bufvec_ut);
-	c2_ut_add(&xcode_ut);
 	c2_ut_add(&xcode_ff2c_ut);
+	c2_ut_add(&xcode_ut);
 	/* These tests have redirection of messages. */
 	c2_ut_add(&addb_ut);
 	c2_ut_add(&console_ut);
-- 
1.8.3.2

